backend/api_v2/deployment_helper.py

```diff
@@ -152,7 +154,12 @@ def execute_workflow(
         """
         workflow_id = api.workflow.id
         pipeline_id = api.id
-        execution_id = str(uuid.uuid4())
+        workflow_execution = WorkflowExecutionServiceHelper.create_workflow_execution(
+            workflow_id=workflow_id,
+            single_step=False,
```

Comment:
it's default False. don't pass

Reply:
we should pass mode as mode WorkflowExecution.Mode.QUEUE,

Reply:
@muhammad-ali-e in case of an API deployment call, shouldn't it be executed instantly when the request is made? Changing this anyway

Reply:
Even it's an instant in the POV of an end user, This might be in celery queue and execute depends on the avaialability of worker. So allways async manner.